kura/org.eclipse.kura.ai.triton.server/src/main/java/org/eclipse/kura/ai/triton/server/TritonServerContainerManager.java

```diff
@@ -237,18 +240,27 @@ private ContainerConfiguration createContainerConfiguration() {
         builder.setMemory(this.options.getContainerMemory());
         builder.setCpus(this.options.getContainerCpus());
         builder.setGpus(this.options.getContainerGpus());
+        builder.setRuntime(this.options.getContainerRuntime());
+        builder.setDeviceList(this.options.getDevices());
 
+        Map<String, String> volumes = new HashMap<>();
         if (this.options.isModelEncryptionPasswordSet()) {
-            builder.setVolumes(Collections.singletonMap(this.decryptionFolderPath, TRITON_INTERNAL_MODEL_REPO));
+            volumes.put(this.decryptionFolderPath, TRITON_INTERNAL_MODEL_REPO);
         } else {
-            builder.setVolumes(
-                    Collections.singletonMap(this.options.getModelRepositoryPath(), TRITON_INTERNAL_MODEL_REPO));
+            volumes.put(this.options.getModelRepositoryPath(), TRITON_INTERNAL_MODEL_REPO);
         }
+        if (!this.options.getBackendsPath().isEmpty()) {
+            volumes.put(this.options.getBackendsPath(), TRITON_INTERNAL_BACKENDS_FOLDER);
+        }
+        builder.setVolumes(Collections.unmodifiableMap(volumes));
 
         List<String> entrypointOverride = new ArrayList<>();
         entrypointOverride.add("tritonserver");
         entrypointOverride.add("--model-repository=" + TRITON_INTERNAL_MODEL_REPO);
         entrypointOverride.add("--model-control-mode=explicit");
+        if (!this.options.getBackendsPath().isEmpty()) {
+            entrypointOverride.add("--backend-directory=" + TRITON_INTERNAL_BACKENDS_FOLDER);
+        }
```

Comment:
I'm not sure you actually need this: if I understood correctly you're mounting the host path where the Triton server already expects its backend folder. 🤔

Reply:
I would say that the best approach is to stay consistent with what we already have (i.e. `TRITON_INTERNAL_MODEL_REPO` approach). Therefore this is ok as-is.